src/lib.rs

```diff
@@ -416,7 +396,7 @@ where
     /// state.
     pub unsafe fn assume_init(self) -> Array<T, U> {
         // TODO(tarcieri): use `MaybeUninit::array_assume_init` when stable
-        Array(ptr::read(self.0.as_ptr().cast()))
+        mem::transmute_copy(&self)
```

Comment:
I'm a little unclear if this is actually correct and it could probably use a better explanatory comment